app/settings_desktop.go

```diff
@@ -41,7 +41,11 @@ func watchFile(path string, callback func()) *fsnotify.Watcher {
 	go func() {
 		for event := range watcher.Events {
 			if event.Op&fsnotify.Remove != 0 { // if it was deleted then watch again
-				watcher.Remove(path)
+				err := watcher.Remove(path)
+				if err != nil {
+					fyne.LogError("Failed to stop watching settings file", err)
+					return
```

Comment:
This is the reason that the app test is failing. Removing the `return` call seems to fix things, but I have a feeling that something might be wrong with the underlying code because it looks like we always get an error saying `Fyne error:  Failed to stop watching settings file2020/03/22 09:20:41   Cause: can't remove non-existent inotify watch`.

Reply:
I think you are right this return is not correct. If we can't stop watching a deleted file (entirely probable) we should return to trying again.
Not sure why this is only an issue on this PR though - the others are working as expected. Are you certain nothing has changed?

Reply:
I think you might understand me wrong here. I added that return statement in my PR to handle the error and the reason that it doesn't cause errors in other ones is because of that.

What I tried saying was that it throws an error here even though I don't think it should and we can see it now because we check the error. :wink:

Reply:
Oh, OK. How do you propose fixing this then? We can't merge with a new break ;)

Reply:
I honestly have no idea that might be causing this. Do you want me to just log the error and not return or just not handle the error at all and leave it for another day?

Reply:
Let me know what you think after reading https://github.com/fsnotify/fsnotify/issues/40 and https://github.com/fsnotify/fsnotify/issues/238

Reply:
The issue seems to be specific to Windows and Linux, perhaps we could only throw an error on operating systems that aren’t those?

Reply:
Pushed new commits with the fix. Might need testing on MacOS, but tests are not failing for me anymore at least :)

Reply:
Looks like MacOS tests on travis still fails. Looks like we always get an error no matter the case. I say that we ignore this for now and get back to it when it is fixed upstream.